helpers/browser/driver.js

```diff
@@ -94,80 +131,106 @@ class ChromeProtocol {
     });
   }
 
-  evaluateFunction(fn, url) {
-    var wrappedScriptStr = '(' + fn.toString() + ')()';
-    return this.evaluateString(wrappedScriptStr, url);
+  evaluateFunction(url, fn) {
+    let wrappedScriptStr = '(' + fn.toString() + ')()';
+    return this.evaluateScript(url, wrappedScriptStr);
   }
 
-  evaluateString(scriptStr, url) {
-    var chrome = this._instance;
+  evaluateScript(url, scriptSrc) {
+    return this.getInstance().then(chrome => {
+      // Set up executionContext listener before navigation.
+      let contextListener = ChromeProtocol.getEvaluationContextFor(chrome, url);
```

Comment:
we'll likely want to do something here like with tracing below, beginning listening, then navigating, then waiting on a promise resolved by the context, then running something in that context